### PR TITLE
Purge support

### PR DESCRIPTION
**Pull request by enst**
I'm relativity new to Go. Just a pull request for you to review. 
It works for me though.
Thanks,

### CODE REVIEW

File: database.go [hunk 1]
```diff
@@ -85,6 +85,23 @@ func (db *Database) Delete(doc CouchDoc) (*DocumentResponse, error) {
 	return newDocumentResponse(body)
 }
 
+// Purge document.
+func (db *Database) Purge(doc CouchDoc) (*DocumentResponse, error) {
+	url := fmt.Sprintf("%s%s", db.URL, "_purge")
+	purgeDoc := map[string]interface{}{doc.GetID(): []string{doc.GetRev()}}
```
Comment (zemirco) [inline database.go]: As far as I know the request is simply `map[string][]string` where the keys are document ids and the values are arrays of document revisions.

**Comment by zemirco** [comment]
Please add some tests for your code.

**Comment by bingli1000** [comment]
map[string][]string  works too. 
But I haven't figured out how to make the test case work.
It seems DocumentResponse does't meet the needs in this case. Not sure if a new type of response struct is needed.

**Comment by zemirco** [comment]
You're right. We'd have to add another response struct that looks like the CouchDB response.

```json
{
    "purge_seq":3,
    "purged":{
        "c6114c65e295552ab1019e2b046b10e": [
            "3-b06fcd1c1c9e0ec7c480ee8aa467bf3b"
        ]
    }
}
```

**Pull request closed without merging**